venv.go

```diff
@@ -202,16 +202,28 @@ func (c VenvCommand) Run() VenvCommandRunOutput {
 			CommandOutput.Error = errors.Wrap(err, "unable to start command")
 			return CommandOutput
 		}
+		
+		output := map[io.ReadCloser]*string{
+			stdout: &CommandOutput.Stdout,
+			stderr: &CommandOutput.Stderr,
+		}
 
-		for _, stream := range []io.ReadCloser{stdout, stderr} {
-			go func(s io.ReadCloser) {
+		for stream, out := range output {
+			go func(s io.ReadCloser, o *string) {
+				var buff bytes.Buffer
+				
 				scanner := bufio.NewScanner(s)
 				scanner.Split(bufio.ScanLines)
+
 				for scanner.Scan() {
 					m := scanner.Text()
 					fmt.Println(m)
+					buff.WriteString(m)
+					buff.WriteString("\n")
 				}
-			}(stream)
+				*o = fmt.Sprint(buff.String())
```

Comment:
This step could have concurrency issue and race condition with line 248,249 when the main routine dumps the streams into the output strings.

Have you considered https://pkg.go.dev/io#TeeReader? It might need some plumbing to get it work, but using a buffer to collect output is probably much safer than dealing with strings in a multithread context. 

In fact, if I had time, I would consider making `CommandOutput.Stderr` and `CommandOutput.Stdout` buffers, so we can write to the output files once a while to avoid empty log files when ansible puller crashed, but this would be for future.

Reply:
<del>The real issue is on the string write part on CommandOutput.Stderr and outString. So in the modified implementation, the concurrency issue still exists. Sorry that my original comment on the `streamOutput` function was a bit misleading.</del>

Actually, reading this part of code more closely, I realize the previous statement about race condition isn't correct. This part actually branches and ends when `c.StreamOutput` is set, so `CommandOutput.Stderr` and `CommandOutput.Stdout` won't be touched by two routines. Sorry for the confusion here caused by the bad readability of the legacy code.

Reply:
There is a minor point about ignoring the errors returned by `cmd.StdoutPipe()` and `cmd.StderrPipe()`, but this is an existing issue in the code, and I'm fine with leaving it out of the scope of this PR.